openupgrade_scripts/scripts/hr_expense/15.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,57 @@
+from openupgradelib import openupgrade
+
+
+def _compute_payment_state(env):
+    if not openupgrade.column_exists(
+        env.cr,
+        "hr_expense_sheet",
+        "payment_state",
+    ):
+        openupgrade.logged_query(
+            env.cr,
+            """
+            ALTER TABLE hr_expense_sheet
+            ADD COLUMN payment_state varchar
+            """,
+        )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE hr_expense_sheet
+        SET payment_state = CASE
+            WHEN am IS NOT NULL AND am.payment_state IS NOT NULL THEN am.payment_state
+            ELSE 'not_paid'
+            END
+        FROM
+            hr_expense_sheet AS hes
+            LEFT JOIN account_move AS am
+                ON hes.account_move_id = am.id
+        WHERE hes.payment_state IS NULL
+        """,
+    )
+
+
+def _compute_currency_id_has_not_value(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE hr_expense
+            SET currency_id = rc.id
+            FROM
+                hr_expense AS he
+                INNER JOIN res_company AS cp
+                    ON cp.id = he.company_id
+                INNER JOIN res_currency AS rc
+                    ON rc.id = cp.currency_id
+            WHERE
+                he.product_id IS NOT NULL
+                AND he.unit_amount IS NOT NULL
+                AND he.currency_id IS NULL
```

Comment:
@daonamutc1 như này thôi là đủ chứ anh?
Ngoài ra, `currency_id` là required trên 15.0, thì phải điến tất cả những dữ liệu chưa có
nếu where thêm cả điều kiện khác, thì có khả năng thiếu trường hợp 
```suggestion
            UPDATE hr_expense e
            SET currency_id = c.currency_id
            FROM res_company c 
            WHERE e.currency_id IS NULL AND c.id = e.company_id
```